### DISCUSSION TITLE
follow_line_newmanager Setup: seeking community tips

### DISCUSSION THREAD

**Discussion by vinayakgavariya** [JdeRobot/RoboticsAcademy#2387] [General]
hello,

i followed the instructions in user guide and successfully done with the set up of the exercise follow_line_newmanager. although i faced some issues and errors but yeah, finally it's done.
i would like to ask for suggestions from the community on what can be done next and how i can complete the exercise in the most efficient way.

kudos to @jmplaza for helpful links!
![image](https://github.com/JdeRobot/RoboticsAcademy/assets/77708101/aaa783e9-b178-487e-ae9e-85350e92f6a9)

**Comment by jmplaza** [comment]
Glad to see you have successfully launched the FollowLine exercise.

You have user docs and tips for solving that exercise [here](https://jderobot.github.io/RoboticsAcademy/exercises/AutonomousCars/follow_line/). That info page describes the available programming interface for that exercise. The API includes the **_hal.py_** module (which provides access to robot sensors and actuators, the HAL, Hardware Abstraction Layer), with the methods:

- HAL.getImage() - to get the image
- HAL.setV() - to set the linear speed
- HAL.setW() - to set the angular velocity

It also includes the _**gui.py**_ module (which provides some visualization methods for debugging such as the GUI.showImage() method).

In fact, each RoboticsAcademy exercise has its own info webpage, available from this URL: [https://jderobot.github.io/RoboticsAcademy/exercises/](https://jderobot.github.io/RoboticsAcademy/exercises/)


Cheers!

**Reply by vinayakgavariya** [reply]
Thanks @jmplaza, I really appreciate it!

I will check-out the links you have shared.

Looking forward to contribute!

**Comment by vinayakgavariya** [comment]
Hey @jmplaza 

I went through the guide maintained by Pankhuri and Sakshay!

It's good and I got bit of clarity too.

But it's the first time I am working on something in Robotics and it's bit overwhelming for me so can you please guide me how can I solve it.

**Comment by jmplaza** [comment]
I'm sorry @vinayakgavariya . Take it easy, and follow some tutorials about basic OpenCV programming and about basic ROS robot programming. There are many available on internet, just google them.

Cheers!

**Comment by vinayakgavariya** [comment]
Sure @jmplaza, I will do that.

**Comment by BkPankaj** [comment]
Hello everyone,
I am facing issues while trying to open the "follow_line_newmanager". It is stuck on the "Connecting" stage and when launching exercises. Could anyone please help me by providing documentation or suggesting a solution? Below are some images for reference.

Connecting and launching error:
![](https://github.com/JdeRobot/RoboticsAcademy/assets/62588059/75802754-4aaf-4131-a507-edddcf3c541c)

Invalid roslaunch XML:
![](https://github.com/JdeRobot/RoboticsAcademy/assets/62588059/325e387a-df92-4328-915b-ebe586e5b08d)

Docker run log:
![](https://github.com/JdeRobot/RoboticsAcademy/assets/62588059/127cc76a-6c7e-4d4a-9cae-bd059c10766b)

**Reply by BkPankaj** [reply]
Greeting everyone,
Does anyone have any idea about this issue? Am I missing something? On my laptop, I only have ROS2 Humble. I'm encountering an issue regarding roslaunch XML and connecting box.

**Reply by OscarMrZ** [reply]
Hi @BkPankaj. There were some issues with the las ROS 1 backend, it is fixed now in RADI version 3.4.29. ROS 2 backend should also work without problems! Please let me now if you encounter further issues.

**Reply by BkPankaj** [reply]
Hi @OscarMrZ sir,
Thank you for your response, and I apologize for the delayed reply. I've observed that when I run RADI with the server, I don't encounter any errors. However, whenever I follow the developer documentation, I encounter errors with RADI version 3.4.29.

**Comment by jmplaza** [comment]
Uhm... that should work. 

You are using RADI 3.4.28, which is based on ROS1 Noetic. It is not relevant at all the ROS release you have got installed locally, as the Gazebo simulator and the robotics application are run inside the RADI (Robotics Application Docker Image).  Try with RADI 3.4.27 or 3.4.26, which can also be downloaded from [DockerHub](https://hub.docker.com/r/jderobot/robotics-academy/tags).

Are you using the RoboticsAcademy from the master branch? try with the [RoboticsAcademy 5.1 release](https://github.com/JdeRobot/RoboticsAcademy/releases/tag/v5.1.0).

Anyway that should work, we will take a look at it.

**Reply by BkPankaj** [reply]
Hello @jmplaza , @OscarMrZ ,

The issue remains unresolved, and I'm uncertain about where I might be going wrong. I have used RoboticsAcademy 5.1 and RADI 3.4.29 as per your recommendation. I would greatly appreciate any assistance anyone can provide in resolving this matter. 

Thank you for your attention to this issue.

**Reply by jmplaza** [reply]
I guess you are just **using** RoboticsAcademy (RA), you are not following the RoboticsAcademy _developer doc_, just the _user doc_ from the [exercise doc page](https://jderobot.github.io/RoboticsAcademy/exercises/AutonomousCars/follow_line/). I've just checked that with RADI 3.4.29 and the follow line exercise works fine...

![followline-3 4 29](https://github.com/JdeRobot/RoboticsAcademy/assets/10961573/5e556cbd-22b7-4a88-b0a9-07d762bd318c)


In case you are following the _developer doc_, then the problem may be related to  [this issue](https://github.com/JdeRobot/RoboticsAcademy/issues/2407). In such a case, a new Pull Request which solves it is coming. Do you have ROS installed locally on your machine?

**Reply by BkPankaj** [reply]
Hello @jmplaza sir, thank you for your reply, and my apologies for the delayed reply. I was caught up with coursework and exams. 
Yes, I have ROS2 Humble installed on my local machine. What I've noticed is that when I follow the developer documentation, I encounter errors regardless of the RADI version I try—whether it's 3.4.29 or 3.4.26. I've also attempted both the master branch and the 5.1 release. However, when I follow the exercise page instructions and run the exercise with the RADI server, I don't encounter any errors.

**Reply by jmplaza** [reply]
Yes, the developer workflow is broken. We are now pretty busy on the upcoming RoboticsAcademy 5.2 release, which has significant changes. Once we publish it we will refine the developer workflow and its documentation. Just wait a little bit. That 5.2 release will be the baseline for the RoboticsAcademy projects in GSoC-2024.

**Reply by BkPankaj** [reply]
Thank you for the update, sir. I am excited and eagerly awaiting the release of RoboticsAcademy 5.2. Additionally, I would like to inquire if there will be any updates regarding the challenges mentioned in the GForm for this year?

Inside docs written GSOC 2022
![](https://github.com/JdeRobot/RoboticsAcademy/assets/62588059/74bf5d58-8f34-407f-9a2f-2c4c6220d90a)

Reactjs challenge is not mentioned above
![](https://github.com/JdeRobot/RoboticsAcademy/assets/62588059/6c274001-89eb-4f78-8900-04940401e4b0)

**Reply by jmplaza** [reply]
Hi @BkPankaj ,

we will not use the same challenges as 2023 nor 2022. We will update them very soon for GSoC-2024. Stay tuned! In addition, this year we will take much into account previous interactions, PRs and issues solved at our software repositories by the contributors in our candidate selection process.

**Reply by BkPankaj** [reply]
Thank you for the update @jmplaza sir
